lib/ansible/modules/packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,412 @@
+                    repos[-1].__setattr__(key, value)
+                elif key in ['Enabled']:
+                    # Use a real boolean
+                    repos[-1].__setattr__(key, value == '1')
+                elif repos:
+                    # Associate value with most recently recorded repo
+                    repos[-1].__setattr__(key, value)
+
+        return repos
+
+    def filter(self, state):
+        '''
+            Return a list of Repositories where state matches (null: all, enabled, disabled)
+        '''
+        if state is None or state == 'all':
+            for repo in self.repos:
+                yield repo
+
+        elif state == 'enabled':
+            for repo in self.repos:
+                if repo.Enabled:
+                    yield repo
+
+        elif state == 'disabled':
+            for repo in self.repos:
+                if not repo.Enabled:
+                    yield repo
+
+    def list_stuff(self, stuff):
+        l={}
+        for repo in self.filter(stuff):
+            id = repo.id
+            id = id.replace("-", "_")
+            l[id] = dict(id=repo.id, name=repo.name, enabled=repo.is_enabled, url=repo.url)
+        return l
+
+    def enable_only(self, id):
+        changed = False
+        to_enable = []
+        to_disable = []
+
+        # no repo id => disable all
+        if len(id) == 0:
+            self.disable_all()
+            changed = True
+            return [changed, dict(enabled=[], disabled='*')]
+
+        # where are we now?
+        current_repos=self.list_stuff('all')
+
+        notice("redhat_repository: enable_only: changed=%s" % str(current_repos))
+        for r in current_repos.keys():
+            notice("redhat_repository: enable_only: **loop** id=%s r=%s" % (str(id), str(r)))
+
+            if current_repos[r]['enabled'] and current_repos[r]['id'] not in id:
+                to_disable.append(current_repos[r]['id'])
+                changed = True
+
+            if not current_repos[r]['enabled'] and current_repos[r]['id'] in id:
+                to_enable.append(current_repos[r]['id'])
+                changed = True
+
+        notice("redhat_repository: enable_only: changed=%s" % str(changed))
+        notice("redhat_repository: enable_only: to_enable=%s" % str(to_enable))
+        notice("redhat_repository: enable_only: to_disable=%s" % str(to_disable))
+
+        if changed:
+            args = "subscription-manager repos "
+            for n in to_disable:
+                args += " --disable=" + n
+
+            for n in to_enable:
+                args += " --enable=" + n
+
+            notice("redhat_repository: enable_only repos args: %s" % str(args))
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+        return [changed, dict(enabled=to_enable, disabled=to_disable)]
+
+    def enablerepo(self, id):
+        if len(id) > 0:
+            args = "subscription-manager repos "
```

Comment:
Indeed. If only I wouldn't have a day job meeting customers and selling our products, I would have the time and oppportunity to increase my Python skills ;)
I wrote this module because it was obviously missing handling RHEL systems. Spending more time is a no-go for me.

And using the Python module should not be the goal, IMHO. The real goal would be to coalesce this module and its sibbling into one, idempotent, ansible module to handle all registration & subscription management for RHEL systems.

I'd suggest we go steps by steps: publish this one, and thereafter call for a revamp of both modules into one, using the RHSM Python module into a new one, with tests, deprecating both this one and the other. Agile style.